### PR TITLE
Use laravel Cache driver in place of file and redis code

### PR DESCRIPTION
**Pull request by cjcox17**
Laravel has a built in Cache class that will allows us to store and fetch information without having to directly access the Storage/Redis/Memcache/Database classes defaulting to which ever driver the user has set in their .env file.

This cleans up the code a bit, removes a unnecessary custom implementation. Also contains some PSR2 fixes from php-cs-fixer(sorry about that).

### CODE REVIEW

**Comment by cjcox17** [comment]
Additional fix. There was a case where an exception would be thrown if sessionData was empty, however it will always be empty if never run before, thus always throwing the exception instead of falling down to the next if statement and getting a new data from the API and setting it in the cache.

**Comment by cjcox17** [comment]
I found myself painstakingly writing large crazy foreach loops with variables tracking the first occurrence of adding a conditional statement to the end of a query string to properly format the query for several fields from an incoming form so I added a query builder.

It essentially wraps the query function using the Laravel query builder to build a query and not execute it, then it grabs the bindings, replacing the placeholders giving us a usable formatted query, then returns the query function with the generated query string.

Updated the README.md to reflect the changes in this pull request.

**Comment by cjcox17** [comment]
Search function was a redundant and quite frankly, overkill. It now accepts an instance of Laravels QueryBuilder. You can build a query using the query builder and pass it through, at which point everything else is the same.

Updated readme to reflect the changes.

File: src/Vtiger.php [hunk 1]
```diff
@@ -618,8 +578,7 @@ protected function _processResponseError($processedData)
         if (!isset($processedData->error)) {
             throw VtigerError::init($this->vTigerErrors, 3);
         }
-
+        dd($processedData);
```
Comment (Chris-Pratt-Clystnet) [inline src/Vtiger.php]: Dont think this should of been left in lol

**Review by Chris-Pratt-Clystnet — CHANGES_REQUESTED** [review]
You left a dd() in the code but other then that these changes look good. Remove the dd and I will look at doing a merge

File: src/Vtiger.php (hunk 1 above)
Comment (cjcox17) [reply src/Vtiger.php]: Whoops, seems I left that in there from when I was debugging! Nice catch haha

**Comment by cjcox17** [comment]
Removed the dd().

**Review by Chris-Pratt-Clystnet — APPROVED** [review]

**Pull request merged**